<h1>Palmer Pengiun Data Analytics</h1>
<hr>
<h3>Analyst: Agustines, Prince C.</h3>

In [1]:
import numpy as np
import pandas as pd
import warnings
warnings.filterwarnings('ignore')

<p>Importing dataset to Jupyter Notebook</p>
<p>Converting dataset to dataframe</p>
<ul><strong>Requirements<strong>
    <li>Name of the Dataframe</li>
    <li>Source</li>
</ul>

In [2]:
penguins = pd.read_csv('DATASETS\\penguins.csv')

In [3]:
penguins

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0       Adelie  Torgersen            39.1           18.7              181.0   
1       Adelie  Torgersen            39.5           17.4              186.0   
2       Adelie  Torgersen            40.3           18.0              195.0   
3       Adelie  Torgersen             NaN            NaN                NaN   
4       Adelie  Torgersen            36.7           19.3              193.0   
..         ...        ...             ...            ...                ...   
339  Chinstrap      Dream            55.8           19.8              207.0   
340  Chinstrap      Dream            43.5           18.1              202.0   
341  Chinstrap      Dream            49.6           18.2              193.0   
342  Chinstrap      Dream            50.8           19.0              210.0   
343  Chinstrap      Dream            50.2           18.7              198.0   

     body_mass_g     sex  year  
0         3750.0    male  2007  
1         3800.0  female  2007  
2         3250.0  female  2007  
3            NaN     NaN  2007  
4         3450.0  female  2007  
..           ...     ...   ...  
339       4000.0    male  2009  
340       3400.0  female  2009  
341       3775.0    male  2009  
342       4100.0    male  2009  
343       3775.0  female  2009  

[344 rows x 8 columns]

<h3>Descriptive Analytics</h3>

In [4]:
penguins.describe()

bill_length_mm  bill_depth_mm  flipper_length_mm  body_mass_g  \
count      342.000000     342.000000         342.000000   342.000000   
mean        43.921930      17.151170         200.915205  4201.754386   
std          5.459584       1.974793          14.061714   801.954536   
min         32.100000      13.100000         172.000000  2700.000000   
25%         39.225000      15.600000         190.000000  3550.000000   
50%         44.450000      17.300000         197.000000  4050.000000   
75%         48.500000      18.700000         213.000000  4750.000000   
max         59.600000      21.500000         231.000000  6300.000000   

              year  
count   344.000000  
mean   2008.029070  
std       0.818356  
min    2007.000000  
25%    2007.000000  
50%    2008.000000  
75%    2009.000000  
max    2009.000000

In [5]:
penguins.describe().T

count         mean         std     min       25%      50%  \
bill_length_mm     342.0    43.921930    5.459584    32.1    39.225    44.45   
bill_depth_mm      342.0    17.151170    1.974793    13.1    15.600    17.30   
flipper_length_mm  342.0   200.915205   14.061714   172.0   190.000   197.00   
body_mass_g        342.0  4201.754386  801.954536  2700.0  3550.000  4050.00   
year               344.0  2008.029070    0.818356  2007.0  2007.000  2008.00   

                      75%     max  
bill_length_mm       48.5    59.6  
bill_depth_mm        18.7    21.5  
flipper_length_mm   213.0   231.0  
body_mass_g        4750.0  6300.0  
year               2009.0  2009.0

<h3>Cleaning the Data Frame</h3>

In [6]:
penguins.isnull().sum()

species               0
island                0
bill_length_mm        2
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           2
sex                  11
year                  0
dtype: int64

<h3>Selecting a column</h3>
<h3>syntax</h3><br>
data_frame['column_name']

In [7]:
penguins['island']

0      Torgersen
1      Torgersen
2      Torgersen
3      Torgersen
4      Torgersen
         ...    
339        Dream
340        Dream
341        Dream
342        Dream
343        Dream
Name: island, Length: 344, dtype: object

In [8]:
penguins['island'] == 'Dream'

0      False
1      False
2      False
3      False
4      False
       ...  
339     True
340     True
341     True
342     True
343     True
Name: island, Length: 344, dtype: bool

In [9]:
penguins[penguins['island'] == 'Dream']

species island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
30      Adelie  Dream            39.5           16.7              178.0   
31      Adelie  Dream            37.2           18.1              178.0   
32      Adelie  Dream            39.5           17.8              188.0   
33      Adelie  Dream            40.9           18.9              184.0   
34      Adelie  Dream            36.4           17.0              195.0   
..         ...    ...             ...            ...                ...   
339  Chinstrap  Dream            55.8           19.8              207.0   
340  Chinstrap  Dream            43.5           18.1              202.0   
341  Chinstrap  Dream            49.6           18.2              193.0   
342  Chinstrap  Dream            50.8           19.0              210.0   
343  Chinstrap  Dream            50.2           18.7              198.0   

     body_mass_g     sex  year  
30        3250.0  female  2007  
31        3900.0    male  2007  
32        3300.0  female  2007  
33        3900.0    male  2007  
34        3325.0  female  2007  
..           ...     ...   ...  
339       4000.0    male  2009  
340       3400.0  female  2009  
341       3775.0    male  2009  
342       4100.0    male  2009  
343       3775.0  female  2009  

[124 rows x 8 columns]

In [10]:
penguins.isnull().sum()

species               0
island                0
bill_length_mm        2
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           2
sex                  11
year                  0
dtype: int64

In [11]:
penguins['island'] == 'Dream'

0      False
1      False
2      False
3      False
4      False
       ...  
339     True
340     True
341     True
342     True
343     True
Name: island, Length: 344, dtype: bool

In [12]:
Adelie_df = penguins[penguins['species'] == 'Adelie']

In [13]:
Adelie_df

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0    Adelie  Torgersen            39.1           18.7              181.0   
1    Adelie  Torgersen            39.5           17.4              186.0   
2    Adelie  Torgersen            40.3           18.0              195.0   
3    Adelie  Torgersen             NaN            NaN                NaN   
4    Adelie  Torgersen            36.7           19.3              193.0   
..      ...        ...             ...            ...                ...   
147  Adelie      Dream            36.6           18.4              184.0   
148  Adelie      Dream            36.0           17.8              195.0   
149  Adelie      Dream            37.8           18.1              193.0   
150  Adelie      Dream            36.0           17.1              187.0   
151  Adelie      Dream            41.5           18.5              201.0   

     body_mass_g     sex  year  
0         3750.0    male  2007  
1         3800.0  female  2007  
2         3250.0  female  2007  
3            NaN     NaN  2007  
4         3450.0  female  2007  
..           ...     ...   ...  
147       3475.0  female  2009  
148       3450.0  female  2009  
149       3750.0    male  2009  
150       3700.0  female  2009  
151       4000.0    male  2009  

[152 rows x 8 columns]

In [14]:
Chinstrap_df = penguins[penguins['species'] == 'Chinstrap']

In [15]:
Chinstrap_df

species island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
276  Chinstrap  Dream            46.5           17.9              192.0   
277  Chinstrap  Dream            50.0           19.5              196.0   
278  Chinstrap  Dream            51.3           19.2              193.0   
279  Chinstrap  Dream            45.4           18.7              188.0   
280  Chinstrap  Dream            52.7           19.8              197.0   
..         ...    ...             ...            ...                ...   
339  Chinstrap  Dream            55.8           19.8              207.0   
340  Chinstrap  Dream            43.5           18.1              202.0   
341  Chinstrap  Dream            49.6           18.2              193.0   
342  Chinstrap  Dream            50.8           19.0              210.0   
343  Chinstrap  Dream            50.2           18.7              198.0   

     body_mass_g     sex  year  
276       3500.0  female  2007  
277       3900.0    male  2007  
278       3650.0    male  2007  
279       3525.0  female  2007  
280       3725.0    male  2007  
..           ...     ...   ...  
339       4000.0    male  2009  
340       3400.0  female  2009  
341       3775.0    male  2009  
342       4100.0    male  2009  
343       3775.0  female  2009  

[68 rows x 8 columns]

In [16]:
penguins[(penguins['species'] == 'Chinstrap') & (penguins['body_mass_g'] >= 4000)]

species island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
284  Chinstrap  Dream            46.0           18.9              195.0   
289  Chinstrap  Dream            52.0           18.1              201.0   
291  Chinstrap  Dream            50.5           19.6              201.0   
295  Chinstrap  Dream            49.2           18.2              195.0   
301  Chinstrap  Dream            52.0           19.0              197.0   
305  Chinstrap  Dream            52.8           20.0              205.0   
307  Chinstrap  Dream            54.2           20.8              201.0   
309  Chinstrap  Dream            51.0           18.8              203.0   
313  Chinstrap  Dream            52.0           20.7              210.0   
315  Chinstrap  Dream            53.5           19.9              205.0   
321  Chinstrap  Dream            50.8           18.5              201.0   
323  Chinstrap  Dream            49.0           19.6              212.0   
329  Chinstrap  Dream            50.7           19.7              203.0   
333  Chinstrap  Dream            49.3           19.9              203.0   
339  Chinstrap  Dream            55.8           19.8              207.0   
342  Chinstrap  Dream            50.8           19.0              210.0   

     body_mass_g     sex  year  
284       4150.0  female  2007  
289       4050.0    male  2007  
291       4050.0    male  2007  
295       4400.0    male  2007  
301       4150.0    male  2007  
305       4550.0    male  2008  
307       4300.0    male  2008  
309       4100.0    male  2008  
313       4800.0    male  2008  
315       4500.0    male  2008  
321       4450.0    male  2009  
323       4300.0    male  2009  
329       4050.0    male  2009  
333       4050.0    male  2009  
339       4000.0    male  2009  
342       4100.0    male  2009

In [17]:
Chinstrap_heavy = penguins[(penguins['species'] == 'Chinstrap') & (penguins['body_mass_g'] >= 4000)]

In [18]:
Chinstrap_heavy

species island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
284  Chinstrap  Dream            46.0           18.9              195.0   
289  Chinstrap  Dream            52.0           18.1              201.0   
291  Chinstrap  Dream            50.5           19.6              201.0   
295  Chinstrap  Dream            49.2           18.2              195.0   
301  Chinstrap  Dream            52.0           19.0              197.0   
305  Chinstrap  Dream            52.8           20.0              205.0   
307  Chinstrap  Dream            54.2           20.8              201.0   
309  Chinstrap  Dream            51.0           18.8              203.0   
313  Chinstrap  Dream            52.0           20.7              210.0   
315  Chinstrap  Dream            53.5           19.9              205.0   
321  Chinstrap  Dream            50.8           18.5              201.0   
323  Chinstrap  Dream            49.0           19.6              212.0   
329  Chinstrap  Dream            50.7           19.7              203.0   
333  Chinstrap  Dream            49.3           19.9              203.0   
339  Chinstrap  Dream            55.8           19.8              207.0   
342  Chinstrap  Dream            50.8           19.0              210.0   

     body_mass_g     sex  year  
284       4150.0  female  2007  
289       4050.0    male  2007  
291       4050.0    male  2007  
295       4400.0    male  2007  
301       4150.0    male  2007  
305       4550.0    male  2008  
307       4300.0    male  2008  
309       4100.0    male  2008  
313       4800.0    male  2008  
315       4500.0    male  2008  
321       4450.0    male  2009  
323       4300.0    male  2009  
329       4050.0    male  2009  
333       4050.0    male  2009  
339       4000.0    male  2009  
342       4100.0    male  2009

<h1>FILL UP NULL VALUES<h1>

In [19]:
penguins.isnull().sum()

species               0
island                0
bill_length_mm        2
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           2
sex                  11
year                  0
dtype: int64

In [20]:
penguins['bill_length_mm'].mean()

np.float64(43.9219298245614)

In [21]:
penguins['bill_length_mm'].fillna(penguins['bill_length_mm'].mean())

0      39.10000
1      39.50000
2      40.30000
3      43.92193
4      36.70000
         ...   
339    55.80000
340    43.50000
341    49.60000
342    50.80000
343    50.20000
Name: bill_length_mm, Length: 344, dtype: float64

In [22]:
penguins['sex'] = penguins['sex'].fillna(penguins['sex'].mode()[0])

In [23]:
penguins.isnull().sum()

species              0
island               0
bill_length_mm       2
bill_depth_mm        2
flipper_length_mm    2
body_mass_g          2
sex                  0
year                 0
dtype: int64

In [24]:
penguins['sex'].mode()

0    male
Name: sex, dtype: object

In [25]:
penguins['bill_length_mm'] = penguins['bill_length_mm'].fillna(penguins['bill_length_mm'].mode()[0])

In [26]:
penguins.isnull().sum()

species              0
island               0
bill_length_mm       0
bill_depth_mm        2
flipper_length_mm    2
body_mass_g          2
sex                  0
year                 0
dtype: int64

In [27]:
penguins['bill_depth_mm'] = penguins['bill_depth_mm'].fillna(penguins['bill_depth_mm'].mode()[0])

In [28]:
penguins.isnull().sum()

species              0
island               0
bill_length_mm       0
bill_depth_mm        0
flipper_length_mm    2
body_mass_g          2
sex                  0
year                 0
dtype: int64

In [29]:
penguins['flipper_length_mm'] = penguins['flipper_length_mm'].fillna(penguins['flipper_length_mm'].mode()[0])

In [30]:
penguins.isnull().sum()

species              0
island               0
bill_length_mm       0
bill_depth_mm        0
flipper_length_mm    0
body_mass_g          2
sex                  0
year                 0
dtype: int64

In [31]:
penguins['body_mass_g'] = penguins['body_mass_g'].fillna(penguins['body_mass_g'].mode()[0])

In [32]:
penguins.isnull().sum()

species              0
island               0
bill_length_mm       0
bill_depth_mm        0
flipper_length_mm    0
body_mass_g          0
sex                  0
year                 0
dtype: int64

<H1>CHANGING DATATYPES</H1>

In [33]:
penguins['species'] = penguins['species'].astype('category')
penguins['island'] = penguins['island'].astype('category')
penguins['year'] = penguins['year'].astype('category')
penguins['sex'] = penguins['sex'].astype('category')

In [34]:
penguins.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 344 entries, 0 to 343
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype   
---  ------             --------------  -----   
 0   species            344 non-null    category
 1   island             344 non-null    category
 2   bill_length_mm     344 non-null    float64 
 3   bill_depth_mm      344 non-null    float64 
 4   flipper_length_mm  344 non-null    float64 
 5   body_mass_g        344 non-null    float64 
 6   sex                344 non-null    category
 7   year               344 non-null    category
dtypes: category(4), float64(4)
memory usage: 12.7 KB


<H1>ADDING COLUMN</H1>

In [35]:
penguins['mass_kg'] = np.round(penguins['body_mass_g'] / 1000,1)

penguins

<h1>Exploring Data Analytics</h1>

<h3>1. What is the average bill length?</h3>

In [36]:
penguins['bill_length_mm'].mean()

np.float64(43.90552325581395)

<h4>Insight: The average bill length is 43.90</h4>

<h3>2. What is the average bill depth?</h3>

In [37]:
penguins['bill_depth_mm'].mean()

np.float64(17.150290697674418)

<h4>Insight: The average bill depth is 17.15</h4>

<h3>3. What has the most population of a species?</h3>

In [38]:
penguins['species'].value_counts()

species
Adelie       152
Gentoo       124
Chinstrap     68
Name: count, dtype: int64

<h4>Insight: The Adelie species has the most population</h4>

<h3>4. Is Female penguins are more than male penguins?</h3>

In [39]:
penguins['sex'].value_counts()

sex
male      179
female    165
Name: count, dtype: int64

<h4>Insight: No, female penguins is not more than male penguins</h4>

<h3>5. What is the average flipper length?</h3>

In [40]:
penguins['flipper_length_mm'].mean()

np.float64(200.85174418604652)

<h4>Insight: The average flipper length is 17.15</h4>

<h3>6. How many adelie is in dream island ?</h3>

In [41]:
adelieindream =  penguins.groupby(['species','island']).count()

In [42]:
adelieindream

bill_length_mm  bill_depth_mm  flipper_length_mm  \
species   island                                                        
Adelie    Biscoe                 44             44                 44   
          Dream                  56             56                 56   
          Torgersen              52             52                 52   
Chinstrap Biscoe                  0              0                  0   
          Dream                  68             68                 68   
          Torgersen               0              0                  0   
Gentoo    Biscoe                124            124                124   
          Dream                   0              0                  0   
          Torgersen               0              0                  0   

                     body_mass_g  sex  year  mass_kg  
species   island                                      
Adelie    Biscoe              44   44    44       44  
          Dream               56   56    56       56  
          Torgersen           52   52    52       52  
Chinstrap Biscoe               0    0     0        0  
          Dream               68   68    68       68  
          Torgersen            0    0     0        0  
Gentoo    Biscoe             124  124   124      124  
          Dream                0    0     0        0  
          Torgersen            0    0     0        0

<h4>Insight: 56 adelies are on dream island</h4>

<h3>7. What sex is the heaviest ?</h3>

In [43]:
heavysex =  penguins.groupby(['sex','mass_kg']).count()

In [44]:
heavysex

species  island  bill_length_mm  bill_depth_mm  \
sex    mass_kg                                                   
female 2.7            1       1               1              1   
       2.8            2       2               2              2   
       2.9            5       5               5              5   
       3.0            6       6               6              6   
       3.1            2       2               2              2   
...                 ...     ...             ...            ...   
male   5.6           11      11              11             11   
       5.7            5       5               5              5   
       5.8            6       6               6              6   
       6.0            5       5               5              5   
       6.3            1       1               1              1   

                flipper_length_mm  body_mass_g  year  
sex    mass_kg                                        
female 2.7                      1            1     1  
       2.8                      2            2     2  
       2.9                      5            5     5  
       3.0                      6            6     6  
       3.1                      2            2     2  
...                           ...          ...   ...  
male   5.6                     11           11    11  
       5.7                      5            5     5  
       5.8                      6            6     6  
       6.0                      5            5     5  
       6.3                      1            1     1  

[68 rows x 7 columns]

<h4>Insight: Male is heavier than Female</h4>

<h3>8. Is there a correlation between bill length and body mass?</h3>

In [45]:
masslength = penguins['bill_length_mm'].corr(penguins['body_mass_g'])

In [46]:
masslength

np.float64(0.5957197979602955)

<h4>Insight: Yes, there is a correlation between bill length and body mass</h4>

<h3>9. What is the most common species in the dream island?</h3>

In [47]:
speciesIndream = penguins[penguins['island'] == 'Dream']
speciesIndream['species'].value_counts()

species
Chinstrap    68
Adelie       56
Gentoo        0
Name: count, dtype: int64

<h4>Insight: Chinstrap is the most common species in the dream island</h4>

<h3>10. Is there a correlation between bill length and bill depth?</h3>

In [48]:
lengthdepth = penguins['bill_length_mm'].corr(penguins['bill_depth_mm'])
lengthdepth

np.float64(-0.23463545677388215)

<h4>Insight: No, there is no correlation between bill length and bill depth</h4>

<h3>11. What is the most common species in the biscoe island?</h3>

In [49]:
speciesIndream = penguins[penguins['island'] == 'Biscoe']
speciesIndream['species'].value_counts()

species
Gentoo       124
Adelie        44
Chinstrap      0
Name: count, dtype: int64

<h4>Insight: Gentoo is the most common species in the biscoe island</h4>

<h3>12. What is the most common species in the torgersen island?</h3>

In [50]:
speciesIndream = penguins[penguins['island'] == 'Torgersen']
speciesIndream['species'].value_counts()

species
Adelie       52
Chinstrap     0
Gentoo        0
Name: count, dtype: int64

<h4>Insight: Adelie is the only species in the torgersen island</h4>

<h1>Data Aggregation</h1>

<h3>Ilan species per island ang nakatira?</h3>

In [ ]:
PerIsland = penguins.groupby(['species','island']).count()

In [ ]:
PerIsland

<h1>TESTING CORRELATION</h1>

<H3>Pearson R Correlation</H3>
<hr>
<p>Used for testing correlation between two <strong>NUMERICAL VALUES<strong></p>

In [ ]:
weight_flipper_cor = penguins['body_mass_g'].corr(penguins['flipper_length_mm'])

In [ ]:
weight_flipper_cor